login/pam.c

```diff
@@ -149,16 +149,37 @@ static int glome_authenticate(pam_handle_t *pamh, login_config_t *config,
   free(action);
   action = NULL;
 
-  char *input = NULL;
-  if (pam_prompt(pamh, PAM_PROMPT_ECHO_ON, &input,
-                 "%s%s\nEnter GLOME code obtained from the above link: ",
-                 config->url_prefix, url)) {
-    return failure(EXITCODE_PANIC, error_tag, "pam-prompt");
+  struct pam_conv *conv;
+  if (pam_get_item(pamh, PAM_CONV, (const void **)&conv) != PAM_SUCCESS) {
+    return failure(EXITCODE_PANIC, error_tag, "pam-get-conv");
   }
 
+  const char *template = "GLOME link: %s%s";
+  size_t message_len =
+      strlen(template) + strlen(config->url_prefix) + strlen(url) + 1;
```

Comment:
Length includes `%s%s` here, so maybe subtract that?

Reply:
I sort of hoped I'd get buy with overestimating. But ok, done.